lib/internal/modules/cjs/loader.js

```diff
@@ -1416,10 +1416,13 @@ function loadESMFromCJS(mod, filename) {
     // createRequiredModuleFacade() to `wrap` which is a ModuleWrap wrapping
     // over the original module.
 
-    // We don't do this to modules that don't have default exports to avoid
-    // the unnecessary overhead. If __esModule is already defined, we will
-    // also skip the extension to allow users to override it.
-    if (!ObjectHasOwn(namespace, 'default') || ObjectHasOwn(namespace, '__esModule')) {
+    // We don't do this to modules that are marked as CJS ESM or that
+    // don't have default exports to avoid the unnecessary overhead.
+    // If __esModule is already defined, we will also skip the extension
+    // to allow users to override it.
+    if (namespace.__cjsUnwrapDefault) {
```

Comment:
Should this allow for any truthy value or strictly `true`? The documentation says ` = true` so we should clarify.

Reply:
With the `'module.exports'` check now I've updated this to an own property check.